tests/board-detection.spec.js

```diff
@@ -0,0 +1,55 @@
+var test = require('tape');
+var proxyquire = require('proxyquire');
+
+test('pid board detection', function(t) {
+  t.plan(2);
+  var Avrgirl = proxyquire('../avrgirl-arduino', {
+    'serialport': {
+      list: function(callback) {
+        callback( null, [
+          { comName: '/dev/cu.sierravsp', manufacturer: '', serialNumber: '',
+            pnpId: '', locationId: '', vendorId: '', productId: '' },
+          { comName: '/dev/cu.Bluetooth-Incoming-Port', manufacturer: '',
+            serialNumber: '', pnpId: '', locationId: '', vendorId: '',
+            productId: '' },
+          { comName: '/dev/cu.usbmodem1421', manufacturer: 'Arduino (www.arduino.cc)',
+            serialNumber: '55432333038351F03170', pnpId: '', locationId: '0x14200000',
+            vendorId: '0x2341', productId: '0x0043' }
+        ]);
+      },
+      SerialPort: require('./helpers/mockSerial').SerialPort
+    }
+  });
+  var avrgirl = new Avrgirl({
+    board: 'uno',
+  });
+  t.notOk(avrgirl.options.port, 'no port was passed in');
```

Comment:
I think it's kind of a mix of the two and probably not needed. I want to make sure that the initial state is that no port was specified (in the setup) or defaulted to (by avrgirl), and that a port is autodetected when we flash. I could reword it was 'initialised without a port', but I think it's best to remove.